gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -137,11 +137,8 @@ public IGitCoreLocalBranch getLocalBranch(String branchName) throws GitCoreExcep
 
   @Override
   public List<IGitCoreRemoteBranch> getAllRemoteBranches() throws GitCoreException {
-    List<IGitCoreRemoteBranch> listOfRemotes = List.of();
-    for (var remoteName : getRemotes()) {
-      listOfRemotes = listOfRemotes.appendAll(getRemoteBranches(remoteName));
-    }
-    return listOfRemotes;
+    return getRemotes().toStream().flatMap(remoteName -> Try.of(() -> getRemoteBranches(remoteName)).get())
```

Comment:
No need to convert to `Stream` and back to `List`; `flatMap` is available on `List` as well